src/components/Explorer/ProjectExplorer.tsx

```diff
@@ -107,8 +108,12 @@ export const ProjectExplorer = ({
   const [contextMenuRow, setContextMenuRow] =
     useState<FileExplorerEntry | null>(null)
   const [isRenaming, setIsRenaming] = useState<boolean>(false)
+  const [isCopying, setIsCopying] = useState<boolean>(false)
   const lastIndexBeforeNothing = useRef<number>(-2)
 
+  // Store a path to copy and paste! Works for folders and files
+  const copyToClipBoard = useRef<FileEntry | null>(null)
```

Comment:
I'm curious as to why useRef and not useState here?

Reply:
Oh so recursive rows can just mutate directly without having to pass handlers and state all the way down?
And clipboard state is not displayed anywhere in the UI, so fine that it's outside of react lifecycle?